src/simulator/controllers/map/map_picker.py

```diff
@@ -89,17 +89,14 @@ def pos(self):
 
         # compute & return logical cube position
         x, y, z = self.__cqueue.get_entry(0).getSurfacePoint(self.__map.root)
-        if self.__map.dim == 2 and math.isclose(z, -self.__map.square_mesh.depth, rel_tol=1e-3):
-            return None  # no picking from underneath
-        pos = [max(math.floor(x), 0), max(math.floor(y), 0), max(math.ceil(z), 0)]
-        if pos[0] == len(self.__data):
-            pos[0] -= 1
-        if pos[1] == len(self.__data[pos[0]]):
-            pos[1] -= 1
-        if pos[2] == len(self.__data[pos[0]][pos[1]]):
-            pos[2] -= 1
-        pos = Point(*pos)
-        return pos
+        x, y, z = [max(math.floor(x), 0), max(math.floor(y), 0), max(math.ceil(z), 0)]
```

Comment:
this will def cause problems with 2d map gen no?

Reply:
not sure why it hasnt already

Reply:
NO! We never use 2D points inside the graphics implementation. Only the interface accepts 2D points and is converted to 3D ASAP